frontend/src/metabase-lib/lib/Question.ts

```diff
@@ -1021,11 +1021,16 @@ class QuestionInner {
   }
 
   dependentMetadata(): DependentMetadataItem[] {
-    if (!this.isDataset()) {
-      return [];
-    }
     const dependencies = [];
 
+    const tableId = this.card().table_id;
+    if (tableId != null) {
+      dependencies.push({
+        type: "table",
+        id: tableId,
+      });
+    }
```

Comment:
In which cases a card contains `table_id`? Isn't it something we only keep in `dataset_query`?

Reply:
The `table_id` on card is the base table of a card's query, so if you have a card based on a card based on a card, it'll unravel down to the base query's source table.

This is something that I'm not 100% certain is needed anymore, now that this is based on https://github.com/metabase/metabase/pull/25267 so I'm going to remove it and see what happens.

Reply:
@daltojohnso in case, we need to keep it, should we maybe add a comment explaining this? It seems to be a pretty rare edge case, I'm afraid it can be perceived as a mistake / very outdated API rather than a special case handling